docs/executor_tutorial/google_lifesciences.rst

```diff
@@ -0,0 +1,880 @@
+that we have not customized this file from the basic tutorial to hard code 
+any storage or executor. We will be telling snakemake to change the executor 
+via command line arguments.
+
+.. code:: python
+
+    SAMPLES = ["A", "B"]
+
+    rule all:
+        input:
+            "plots/quals.svg"
+
+    rule bwa_map:
+        input:
+        fastq="samples/{sample}.fastq",
+        idx=multiext("genome.fa", ".amb", ".ann", ".bwt", ".pac", ".sa")
+    conda:
+        "environment.yml"
+    output:
+        "mapped_reads/{sample}.bam"
+    params:
+        idx=lambda w, input: os.path.splitext(input.idx[0])[0]
+    shell:
+        "bwa mem {params.idx} {input.fastq} | samtools view -Sb - > {output}"
+
+    rule bwa_map:
+        input:
+            "genome.fa",
+            "samples/{sample}.fastq"
+        conda:
+            "environment.yml"
+        output:
+            "mapped_reads/{sample}.bam"
+        shell:
+            "bwa mem {input} | samtools view -Sb - > {output}"
+
+    rule samtools_sort:
+        input:
+            "mapped_reads/{sample}.bam"
+        output:
+            "sorted_reads/{sample}.bam"
+        conda:
+            "environment.yml"
+        shell:
+            "samtools sort -T sorted_reads/{wildcards.sample} "
+            "-O bam {input} > {output}"
+
+    rule samtools_index:
+        input:
+            "sorted_reads/{sample}.bam"
+        output:
+            "sorted_reads/{sample}.bam.bai"
+        conda:
+            "environment.yml"
+        shell:
+            "samtools index {input}"
+
+    rule bcftools_call:
+        input:
+            fa="genome.fa",
+            bam=expand("sorted_reads/{sample}.bam", sample=SAMPLES),
+            bai=expand("sorted_reads/{sample}.bam.bai", sample=SAMPLES)
+        output:
+            "calls/all.vcf"
+        conda:
+            "environment.yml"
+        shell:
+            "samtools mpileup -g -f {input.fa} {input.bam} | "
+            "bcftools call -mv - > {output}"
+
+    rule plot_quals:
+        input:
+            "calls/all.vcf"
+        output:
+            "plots/quals.svg"
+        conda:
+            "environment.yml"
+        script:
+            "plot-quals.py"
+
+
+
+And let's also write the script in our present working directory for the last step
+to do the plotting - call this `plot-quals.py`:
+
+.. code:: python
+
+    import matplotlib
+    matplotlib.use("Agg")
+    import matplotlib.pyplot as plt
+    from pysam import VariantFile
+
+    quals = [record.qual for record in VariantFile(snakemake.input[0])]
+    plt.hist(quals)
+
+    plt.savefig(snakemake.output[0])
+
+
+Step 3: Run Snakemake
+:::::::::::::::::::::
+
+Now let's run Snakemake with the Google Life Sciences Executor.
+
+
+.. code:: console
+
+    snakemake --google-lifesciences --default-remote-prefix snakemake-testing-data --use-conda --google-lifesciences-region us-west1 --container-image snakemake/snakemake:v5.10.0
```

Comment:
```suggestion
    snakemake --google-lifesciences --default-remote-prefix snakemake-testing-data --use-conda --google-lifesciences-region us-west1
```

Never tell people to hardcode a container image. Snakemake selects the matching image automatically. If not that's a bug that we should fix ASAP.

Reply:
There was a bug very early on in testing, but I think it's resolved now. I take a more conservative approach to (at least for documentation that I want to be sure is tested and working) hard code versions that I have tested and know to work. But I also agree that we should assume later versions will be working, and not restrict to (what will always become) an older version, likely with fewer features / more bugs. And with this change, if a user does hit an error with a newer image, then they can report it and we fix it!